nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/LookupAttribute.java

```diff
@@ -0,0 +1,289 @@
+@DynamicProperty(name = "The name of the attribute to add to the FlowFile",
+    value = "The name of the key or property to retrieve from the lookup service",
+    supportsExpressionLanguage = true,
+    description = "Adds a FlowFile attribute specified by the dynamic property's key with the value found in the lookup service using the the dynamic property's value")
+@WritesAttribute(attribute = "See additional details", description = "This processor may write zero or more attributes as described in additional details")
+public class LookupAttribute extends AbstractProcessor {
+
+    public static final String MULTI_STRATEGY = "multi-criteria";
+
+    public static final String KEY_VALUE_STRATEGY = "key-value";
+
+    public static final PropertyDescriptor LOOKUP_SERVICE =
+        new PropertyDescriptor.Builder()
+            .name("lookup-service")
+            .displayName("Lookup Service")
+            .description("The lookup service to use for attribute lookups")
+            .identifiesControllerService(StringLookupService.class)
+            .required(true)
+            .build();
+
+    static final PropertyDescriptor LOOKUP_STRATEGY = new PropertyDescriptor.Builder()
+        .name("lookup-strategy")
+        .displayName("Lookup Strategy")
+        .description("The strategy to use to retrieve results from the " +
+            "lookup service. If " + MULTI_STRATEGY + " then the conjunction " +
+            "of all the dynamic properties will be used for coordinate " +
+            "lookups and a destination attribute must be provided. " +
+            "Otherwise, for " + KEY_VALUE_STRATEGY + " each dynamic property " +
+            "specifies a destination attribute name and a target key for " +
+            "lookup.")
+        .allowableValues(MULTI_STRATEGY, KEY_VALUE_STRATEGY)
+        .defaultValue(KEY_VALUE_STRATEGY)
+        .required(true)
+        .build();
+
+    static final PropertyDescriptor DESTINATION_ATTRIBUTE = new PropertyDescriptor.Builder()
+        .name("destination-attribute")
+        .displayName("Destination Attribute")
+        .description("The attribute to store the lookup result in if using " +
+            "the " + MULTI_STRATEGY + " lookup strategy.")
+        .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+        .expressionLanguageSupported(true)
+        .required(false)
+        .build();
+
+    public static final PropertyDescriptor INCLUDE_EMPTY_VALUES =
+        new PropertyDescriptor.Builder()
+            .name("include-empty-values")
+            .displayName("Include Empty Values")
+            .description("Include null or blank values for keys that are null or blank")
+            .addValidator(StandardValidators.BOOLEAN_VALIDATOR)
+            .allowableValues("true", "false")
+            .defaultValue("true")
+            .required(true)
+            .build();
+
+    public static final Relationship REL_SUCCESS = new Relationship.Builder()
```

Comment:
Should this be "matched" instead of "success"? Seems like it better goes with the "unmatched" relationship in terms of behavior.